### PR TITLE
workflows: add build/release workflow

### PR DESCRIPTION
**Pull request by roman-khimov**
And rework release procedure a bit. Don't rely on external tools and make CHANGELOGs better.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+name: Build
+
+on:
+  pull_request:
+    branches:
+      - master
+    types: [opened, synchronize]
+    paths-ignore:
+      - 'scripts/**'
```
Comment (carpawell) [inline .github/workflows/build.yml:9]: we dont have ones but ok

**Review by carpawell — APPROVED** [review]
Have you tested it?

**Comment by roman-khimov** [comment]
> Have you tested it?

There were some fixes, but release can be tested only with a release. It at the same time is very similar to what NeoGo does.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          go-version: '1.21'
+          cache: true
+
+      - name: Build
+        run: make
+        env:
+          GOARCH: ${{ matrix.arch }}
+
+      - name: Rename binaries
+        run: for i in ./bin/*; do mv $i $i-${{ matrix.os.bin-name }}-${{ matrix.arch }}${{ (matrix.os.bin-name == 'windows' && '.exe') || '' }}; done
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply .github/workflows/build.yml:62]: They'll be exactly the same (that's why they're renamed) except for missing .tar.gz. I know .tar.gz is used in some cases, but those can be reworked and this duplication was never good.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.ref }}
+          # Allows to fetch all history for all branches and tags. Need this for proper versioning.
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply .github/workflows/build.yml:53]: Sure, as usual.

**Review by cthulhu-rider — APPROVED** [review]

File: .github/workflows/build.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (carpawell) [reply .github/workflows/build.yml:62]: a new "linux" word in the binary?

[the remark this answers is not in the document]
Comment (roman-khimov) [reply .github/workflows/build.yml:62]: Check the binaries from https://github.com/nspcc-dev/neofs-node/actions/runs/7979125234?pr=2743, do we have correct number of linuxes there?

[the remark this answers is not in the document]
Comment (roman-khimov) [reply .github/workflows/build.yml:62]: Updated CHANGELOG as well, btw.

**Pull request merged**